File: python/src/renting/Renters.py
from src.renting.Renter import Renter
from src.renting.RenterId import RenterId


class Renters:
    _renters: dict[RenterId, Renter] = {}

    def __init__(self, *renters: Renter):
        self._renters = {renter.id: renter for renter in renters}

    def find_by_id(self, item_id: RenterId) -> Renter | None:
        return self._renters.get(item_id, None)

File: java-rent-nullaway/src/main/java/de/richargh/billiondollar/Renters.java
package de.richargh.billiondollar;

import java.util.Map;
import java.util.Optional;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class Renters {

    private final Map<RenterId, Renter> renters;

    public Renters(Renter... renters) {
        this.renters = Stream.of(renters)
                .collect(Collectors.toMap(Renter::id, user -> user));
    }

    public Optional<Renter> findById(RenterId id) {
        return Optional.ofNullable(renters.get(id));
    }
}
